processing/src/main/java/org/apache/druid/segment/UnnestStorageAdapter.java

```diff
@@ -366,24 +366,16 @@ void addPreFilter(@Nullable final Filter filter)
       final int origFilterSize;
       if (queryFilter.getRequiredColumns().contains(outputColumnName)) {
         // outside filter contains unnested column
-        // requires check for OR
-        if (queryFilter instanceof OrFilter) {
-          origFilterSize = ((OrFilter) queryFilter).getFilters().size();
-          for (Filter filter : ((OrFilter) queryFilter).getFilters()) {
-            if (filter.getRequiredColumns().contains(outputColumnName)) {
-              final Filter newFilter = rewriteFilterOnUnnestColumnIfPossible(
-                  filter,
-                  inputColumn,
-                  inputColumnCapabilites
-              );
-              if (newFilter != null) {
-                preFilterList.add(newFilter);
-              }
-            } else {
-              preFilterList.add(filter);
-            }
-          }
-          if (preFilterList.size() == origFilterSize) {
+        // requires check for OR and And filters, disqualify rewrite for non-unnest filters
+        if (queryFilter instanceof BooleanFilter) {
+          origFilterSize = ((BooleanFilter) queryFilter).getFilters().size();
+          recursiveRewriteOnUnnestFilters(
+              (BooleanFilter) queryFilter,
+              preFilterList,
+              inputColumn,
+              inputColumnCapabilites
+          );
+          if (preFilterList.size() >= origFilterSize) {
             // there has been successful rewrites
             final OrFilter preOrFilter = new OrFilter(preFilterList);
             filterSplitter.addPreFilter(preOrFilter);
```

Comment:
why do we always make an `OrFilter` if we are only checking for `BooleanFilter` to break down? Isn't this going to turn `AndFilter` into `OrFilter`? Am I missing something?

Reply:
you are previously this code was dealign with only OR filters, we need to change the final filter based on type. I Need refactor the code to create new PreFilterList.

Reply:
Oh shoot, I missed that. Good catch

Reply:
There seems to be an issue with the test `test_pushdown_or_filters_unnested_and_original_dimension_with_unnest_adapters` as the expected results have changed. I'm going through the code today